plugins/lombok/lombok-compiler-plugin/testData/compile/with.kt

```diff
@@ -0,0 +1,33 @@
+//FILE: WithExample.java
+
+import lombok.AccessLevel;
+import lombok.With;
+
+public class WithExample {
+
+    //todo replace constructors with annotations when supported
+    public WithExample() {
+
+    }
+
+    public WithExample(int age, String name) {
+
+    }
+
+    @With private int age = 10;
+
+    @With private String name;
+
+    static void test() {
+        WithExample obj = new WithExample().withAge(16).withName("fooo");
+    }
+}
+
+
+//FILE: test.kt
+
+class Test {
+    fun run() {
+        val obj: WithExample = WithExample().withAge(16).withName("fooo")
+    }
+}
```

Comment:
I think it's worth having some kind of full pipeline test on this to check if backend generates correct code for such calls

Reply:
run() functions in these tests are actually executed (it is basically sort of box test). Will make it more explicit (add assertions)